lib/razorpay/request.rb

```diff
@@ -80,6 +84,10 @@ def make_test_request
     def create_instance(res)
       response = res.parsed_response
 
+      if response.is_a?(Array)==true && response.length == 0
```

Comment:
If you are specifically checking for empty array then `empty?` instead of length will do. Else, if you are checking if elements inside array are also empty, then you need to reduce it and check.

Reply:
Done